<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro">

  <gazebo>
    <plugin name="ros_control" filename="libgazebo_ros_control.so">
      <controlPeriod> 0.01 </controlPeriod>
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
  </gazebo>
  
    
  <!-- Gazebo grasping plugin -->
  <gazebo>
    <gripper name="gazebo_gripper">
      <grasp_check>
        <attach_steps>2</attach_steps>    <!-- default: 20 -->
        <detach_steps>2</detach_steps>    <!-- default: 40 -->
        <min_contact_count>3</min_contact_count>
      </grasp_check>
      <gripper_link>robotiq_85_left_finger_tip_link</gripper_link>
      <gripper_link>robotiq_85_right_finger_tip_link</gripper_link>
      <palm_link>robotiq_85_base_link</palm_link>
    </gripper>
  </gazebo>

  <xacro:include filename="$(find agimus_demos)/urdf/_d435.gazebo.xacro"/>
  <xacro:gazebo_d435 camera_name="d435" reference_link="ref_camera_link"
                     topics_ns="camera"
                     depth_optical_frame="camera_depth_optical_frame"
                     color_optical_frame="camera_color_optical_frame"
                     infrared1_optical_frame="camera_infrared1_optical_frame"
                     infrared2_optical_frame="camera_infrared2_optical_frame">
  </xacro:gazebo_d435>
</robot>
